File: src/Player.cpp
#include <iostream>
#include <cstring>
#include "LinkedList.cpp"
#include "Player.h"
#include "Cell.h"
#include "Product.h"
using namespace std;

//Constructor
Player::Player(int x, int y, Cell& _c) {
    posX = x;
    posY = y;
    water = 0;  
    money = 0;
    _c.setElement(posX, posY, 'P');
    LinkedList<string> inventory;
}

Player::~Player() {
    // cout<<"THE END"<<endl;
}

void Player::talk() {
    cout<<"Player : Halo"<<endl;
    cout<<"Player now in "<< posX << " " << posY <<endl;

}

void Player::print() {
    cout << "Player Info" << endl;
    cout << "Position: " << posX << ", " << posY << endl;
    cout << "Water: " << water << endl;
    cout << "Money: " << money << endl;
    inventory.print();
}

int Player::getPosX() {
    return posX;
}

int Player::getPosY() {
    return posY;
}

int Player::getMoney() {
    return money;
}

int Player::getWater() {
    return water;
}

void Player::move (Cell& _c) {
    char cc;
    cout << "Select Move : ";
    cin >> cc;
    switch (cc){
        case 'w' :  
            if(posX-1>=0 && posX-1<15 && posY>=0 && posY<15){
                if(_c.getElement(posX-1, posY)==' '){
                    _c.setElement(posX, posY, ' ');
                    posX--;
                    _c.setElement(posX, posY, 'P');
                }else{
                    cout << "Ada Sesuatu" << endl;
                }
            }
            break;
        case 'a' :
            if(posX>=0 && posX<15 && posY-1>=0 && posY-1<15){ 
                if(_c.getElement(posX, posY-1)==' '){ 
                    _c.setElement(posX, posY, ' ');
                    posY--;
                    _c.setElement(posX, posY, 'P');
                }else{
                    cout << "Ada Sesuatu" << endl;
                }
            }
            break;
        case 's' :  
            if(posX+1>=0 && posX+1<15 && posY>=0 && posY<15){
                if(_c.getElement(posX+1, posY)==' '){
                    _c.setElement(posX, posY, ' ');
                    posX++;
                    _c.setElement(posX, posY, 'P');
                }else{
                    cout << "Ada Sesuatu" << endl;
                }
            }
            break;
        case 'd' :  
            if(posX>=0 && posX<15 && posY+1>=0 && posY+1<15){
                if(_c.getElement(posX, posY+1)==' '){
                    _c.setElement(posX, posY, ' ');
                    posY++;
                    _c.setElement(posX, posY, 'P');
                }else{
                    cout << "Ada Sesuatu" << endl;
                }
            }
            break;
        default:
            cout << "Invalid command" << endl;
    }
}

int Player::getInteractType(Facility &_f) {
    if (isAdjacent(_f.getWellX(), _f.getWellY()) ||
        isAdjacent(_f.getMixerX(), _f.getMixerY()) ||
        isAdjacent(_f.getTruckX(), _f.getTruckY())) {
        return 0;
    }

    return 1;
}

bool Player::isAdjacent(int x, int y ) {
    if (((posX + 1) == x) ||
        ((posX - 1) == x) ||
        ((posY + 1) == y) ||
        ((posY - 1) == y)) {
        return true;
    }

    return false;
}

void Player::interact(Facility &_f) {
    int interactType = this->getInteractType(_f);
    if (interactType == 0) {
        this->interactFacility(_f);
    } else {
        // Interact Animal
    }
}

void Player::interactFacility(Facility &_f) {
    if (isAdjacent(_f.getWellX(), _f.getWellY())) {
        water = 10;
    } else if (isAdjacent(_f.getMixerX(), _f.getMixerY())) {
        // Interact Mixed
    } else if (isAdjacent(_f.getTruckX(), _f.getTruckY())) {
        if(inventory.isEmpty()){
            cout << "Inventory Kosong" << endl;
        }else{
            string isi;
            while(!inventory.isEmpty()){
                isi = inventory.get(0);
                inventory.remove(isi);
                if ( strcmp(isi.c_str(),"ChickenMeat")==0 ){
                    money += 4;
                }else if(strcmp(isi.c_str(), "ChickenEgg") == 0){
                    money += 3;
                }else if(strcmp(isi.c_str(), "DuckMeat") == 0){
                    money += 5;
                }else if(strcmp(isi.c_str(), "DuckEgg") == 0){
                    money += 4;
                }else if(strcmp(isi.c_str(), "CowMeat") == 0){
                    money += 6;
                }else if(strcmp(isi.c_str(), "CowMilk") == 0){
                    money += 4;
                }else if(strcmp(isi.c_str(), "GoatMeat") == 0){
                    money += 5;
                }else if(strcmp(isi.c_str(), "GoatMilk") == 0){
                    money += 3;
                }else if(strcmp(isi.c_str(), "LambMeat") == 0){
                    money += 5;
                }else if(strcmp(isi.c_str(), "Omelette") == 0){
                    money += 3;
                }else if(strcmp(isi.c_str(), "Pork") == 0){
                    money += 5;
                }else if(strcmp(isi.c_str(), "Sausage") == 0){
                    money += 5;
                }
            }
            cout << "Truck Succes" << endl;
        }
    }
}

//  //Interact dengan Truck
//  if((((posX + 1) == TruckX) || ((posX - 1) == TruckX)) && (((posY + 1) == TruckY) || ((posY - 1) == TruckY))) {
//      if (LinkedList::isEmpty()) {
//          //Do Nothing / pesan error
//      }
//      else {
//          while !(LinkedList::isEmpty()) {
//              money += SideProduct::getProductValue(LinkedList::get(0));
//              LinkedList::remove(LinkedList::get(0));
//          }
//      }
//  }

//  //Interact dengan Mixer
//  if((((posX + 1) == MixerX) || ((posX - 1) == MixerX)) && (((posY + 1) == MixerY) || ((posY - 1) == MixerY))) {
//      //campur bahannya
//      //masukin inventory
//  }   

//  //Interact dengan FarmAnimal
//  {
//      //ayam(A)
//      if((Cell::getElement(posX+1,posY) == "A") || (Cell::getElement(posX-1,posY) == "A") || (Cell::getElement(posX,posY+1) == "A") || (Cell::getElement(posX,posY-1) == "A"))
//          ChickenEgg::ChickenEgg();
//          inventory.add(chicken::getProduct());
//      //bebek(B)
//      if((Cell::getElement(posX+1,posY) == "B") || (Cell::getElement(posX-1,posY) == "B") || (Cell::getElement(posX,posY+1) == "B") || (Cell::getElement(posX,posY-1) == "B"))
//          DuckEgg::DuckEgg();
//          inventory.add(bebek::getProduct());
//      //kambing(K)
//      if((Cell::getElement(posX+1,posY) == "K") || (Cell::getElement(posX-1,posY) == "K") || (Cell::getElement(posX,posY+1) == "K") || (Cell::getElement(posX,posY-1) == "K"))
//          GoatMilk::GoatMilk();
//          inventory.add(kambing::getProduct());
//      //Sapi(S)
//      if((Cell::getElement(posX+1,posY) == "S") || (Cell::getElement(posX-1,posY) == "S") || (Cell::getElement(posX,posY+1) == "S") || (Cell::getElement(posX,posY-1) == "S"))
//          CowMilk::CowMilk();
//          inventory.add(sapi::getProduct());
//  }
// }

void Player::Kill(Cell &_c/*,  vector<FarmAnimal*>& animals*/) {
 //kill animalnya
 //dapeting dagingnya
    int cc;
    cout << "Select Side to Kill" << endl;
    cout << "1. Atas    2. Bawah" << endl;
    cout << "3. Kiri    4. Kanan" << endl;
    cin >> cc;
    switch(cc){
        case 1 :
            cekhewan(posX-1, posY, _c);
<<<<<<< HEAD
            animals.setStatus(false);
=======
            // vector<FarmAnimal*>::iterator it;
            // for(it = animals.begin(); it!=animals.end(); ++it){
            //     if(animals.posX == this->posX-1 && animals.posY == this->posY){
            //         (*it)->setStatus(0);
            //     }
            // }
>>>>>>> truck, kill belum bisa
            break;
        case 2 :
            cekhewan(posX+1, posY, _c);
            break;
        case 3 :
            cekhewan(posX, posY-1, _c);
            break;
        case 4 :
            cekhewan(posX, posY+1, _c);
            break;
        default :
            cout << "Command Salah" << endl;
    }
     
}

void Player::cekhewan(int x, int y, Cell &_c){
    //ayam(A)
     if(_c.getElement(x,y) == 'A'){
         ChickenMeat a = ChickenMeat();
         inventory.add("ChickenMeat");
        //  ayam::~ayam();
     }
     //bebek(B)
     else if(_c.getElement(x,y) == 'B'){
         DuckMeat a = DuckMeat();
         inventory.add("DuckMeat");
        //  bebek::~bebek();
     } 
     //babi(b)
     else if(_c.getElement(x,y) == 'Z'){
         Pork a = Pork();
         inventory.add("Pork");
        //  babi::~babi();
     } 
     //domba(D)
     else if(_c.getElement(x,y) == 'D'){
         LambMeat a = LambMeat();
         inventory.add("LambMeat");
        //  domba::~domba();
     }
     //kambing(G)
     else if(_c.getElement(x,y) == 'G'){
         GoatMeat a = GoatMeat();
         inventory.add("GoatMeat");
        //  kambing::~kambing();
     }
     //Sapi(C)
     else if(_c.getElement(x,y) == 'C'){
         CowMeat a = CowMeat();
         inventory.add("CowMeat");
        //  sapi::~sapi();
     }else{
         cout << "Kosong" << endl;
     }   
     this->print();
}

void Player::Grow(Cell &_c) {
    if (water > 0) {
        //Mengisi petak cell dengan notasi rumput
        if(_c.getRumput(posX, posY) == ' '){
            _c.setRumput(posX, posY);
            water--;
        }
        else {
            //udah ada rumput, jadi do nothing/pesen error
            cout << "Can't grow anymore!" << endl; 
        }
    }
    else {
        //gaada air, sana pergi ke well
        cout << "insufficient water!" << endl;
    }
    this->print();
}

File: java/src/Player.java
import java.util.*;
/**
 * Player Class
 */
class Player implements LivingThings {
    /** Player X position. */
    int posX;
    /** Player Y position. */
    int posY;
    /** How much water the player carry. */
    int water;
    /** How much money the player own. */
    int money;
    /** Player inventory. */
    Vector<String> inventory;

    /**
     * Player ctor
     *
     * @param x THe initial x position of Player.
     * @param y The initial y position of player.
     * @param c The cell the player refer to.
     */
    public Player(int x, int y, Cell c) {
        posX = x;
        posY = y;
        water = 0;
        money = 0;
        c.setElement(posX, posY, 'P');
        inventory = new Vector<String>();
    }

    /**
     * Talk to Player.
     */
    public void talk() {
        System.out.println("Player : Halo");
        System.out.println("Player now in " + posX + ", " + posY);
    }

    /**
     * print all Player info.
     */
    public void print() {
        System.out.println("Player Info");
        System.out.println("Position : " + posX + ", " + posY);
        System.out.println("Water : " + water);
        System.out.println("Money : " + money);
        System.out.print("Inventory : ");
        // print isi inventory
        System.out.println(inventory);
    }

    /**
     * X position getter.
     *
     * @return int The X position of Player.
     */
    public int getPosX() {
        return posX;
    }

    /**
     * Y position getter.
     *
     * @return int The Y position of Player.
     */
    public int getPosY() {
        return posY;
    }

    /**
     * Money getter.
     *
     * @return int The Player's money.
     */
    public int getMoney() {
        return money;
    }
    
    /**
     * Water getter.
     *
     * @return int The Player's water.
     */
    public int getWater() {
        return water;
    }

    /**
     * Move the player.
     *
     * @param _c The cell the player refer to.
     */
    public void move(Cell _c) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Select Move : ");
        char cc = sc.next().charAt(0); 
        switch (cc) {
            case 'w' :  
                if(posX-1>=0 && posX-1<15 && posY>=0 && posY<15) {
                    if(_c.getElement(posX-1, posY)==' ') {
                        _c.setElement(posX, posY, ' ');
                        posX-=1;
                        _c.setElement(posX, posY, 'P');
                    } else{
                        System.out.println("Ada Sesuatu");
                    }
                }
                break;
            case 'a' :
                if(posX>=0 && posX<15 && posY-1>=0 && posY-1<15) { 
                    if(_c.getElement(posX, posY-1)==' ') { 
                        _c.setElement(posX, posY, ' ');
                        posY-=1;
                        _c.setElement(posX, posY, 'P');
                    } else{
                        System.out.println("Ada Sesuatu");
                    }
                }
                break;
            case 's' :  
                if(posX+1>=0 && posX+1<15 && posY>=0 && posY<15) {
                    if(_c.getElement(posX+1, posY)==' ') {
                        _c.setElement(posX, posY, ' ');
                        posX+=1;
                        _c.setElement(posX, posY, 'P');
                    } else{
                        System.out.println("Ada Sesuatu");
                    }
                }
                break;
            case 'd' :  
                if(posX>=0 && posX<15 && posY+1>=0 && posY+1<15) {
                    if(_c.getElement(posX, posY+1)==' ') {
                        _c.setElement(posX, posY, ' ');
                        posY+=1;
                        _c.setElement(posX, posY, 'P');
                    } else{
                        System.out.println("Ada Sesuatu");
                    }
                }
                break;
            default:
            System.out.println("Invalid command");
        }
    }

    /**
     * Get interaction type done by player.
     *
     * @param _f Facility in board.
     * @param side The side which player interact with.
     * @return Interaction Type, 1 for animal interaction
     *                           0 for facility interaction
     */
    public int getInteractType(Facility _f, int side) {
        if (isAdjacent(_f.getWellX(), _f.getWellY(), side) ||
            isAdjacent(_f.getMixerX(), _f.getMixerY(), side) ||
            isAdjacent(_f.getTruckX(), _f.getTruckY(), side)) {
            return 0;
        }
        return 1;
    }

    /**
     * Check whether a side player choosen is adjacent to something.
     *
     * @param x X position of the object.
     * @param y Y position of the object.
     * @param side The side which player choose.
     * @return True if object is on the side the player choose,
     *         False otherwise
     */
    public boolean isAdjacent(int x, int y, int side) {
        boolean retval = false;
        switch (side) {
            case 1: {
                retval = ((posX - 1) == x && posY == y);    
                break;
            }
    
            case 2: {
                retval = ((posX + 1) == x && posY == y);
                break;
            }
    
            case 3: {
                retval = ((posY - 1) == y && posX == x);
                break;
            }
    
            case 4: {
                retval = ((posY + 1) == y && posX == x);
                break;
            }
        }
        return retval;
    }

    /**
     * Player interaction with an object.
     *
     * @param _f Facility on Board.
     * @param _c The cell thhe Player refer to.
     * @param animals List of animals on Board.
     */
    public void interact(Facility _f, Cell _c, Vector<FarmAnimal> animals) {
        System.out.println("Select Side to Interact");
        System.out.println("1. Atas    2. Bawah");
        System.out.println("3. Kiri    4. Kanan");
        Scanner sc = new Scanner(System.in);
        System.out.print("Select Move : ");
        int cc = sc.nextInt(); 
    
        int interactType = this.getInteractType(_f, cc);
        if (interactType == 0) {
            this.interactFacility(_f, cc);
        } else {
            this.interactAnimal(_c, cc, animals);
        }
    }

    /**
     * Interact with animal.
     *
     * @param _c The cell the Player refer to.
     * @param side The side Player interact with.
     * @param animals List of animals on Board.
     */
    public void interactAnimal(Cell _c, int side, Vector<FarmAnimal> animals) {
         switch (side) {
            case 1: {
                cekAnimal(_c, posX - 1, posY, animals);
                break;
            }
    
            case 2: {
                cekAnimal(_c, posX + 1, posY, animals);
                break;
            }
    
            case 3: {
                cekAnimal(_c, posX, posY - 1, animals);
                break;
            }
    
            case 4: {
                cekAnimal(_c, posX, posY + 1, animals);
                break;
            }
        }
    }

    /**
     * Check animal the player interact with. If there is an animal,
     * the Player get the animal product. Otherwise, the player got nothing.
     *
     * @param _c The cell the Player refer to.
     * @param x The X position of the cell Player interact with.
     * @param y The Y position of the cell Player interact with.
     * @param animals The list of animals on Board.
     */
    public void cekAnimal(Cell _c, int x, int y, Vector<FarmAnimal> animals) {
        if(_c.getElement(x,y) == 'A') {
            for(int i=0; i<animals.size(); i++) {
                if(animals.get(i).getPosX() == x && animals.get(i).getPosY() == y && animals.get(i).getCountProduct()>0) {
                    System.out.println("ayam " + animals.get(i).getCountProduct());
                    animals.get(i).getProduct();
                    inventory.add("ChickenEgg");
                }
            }
         }
         // bebek(B)
         else if(_c.getElement(x,y) == 'B') {
            for(int i=0; i<animals.size(); i++) {
                if(animals.get(i).getPosX() == x && animals.get(i).getPosY() == y && animals.get(i).getCountProduct()>0) {
                    System.out.println("bebek " + animals.get(i).getCountProduct());
                    animals.get(i).getProduct();
                    inventory.add("DuckEgg");
                }
            }
         } 
         // kambing(G)
         else if(_c.getElement(x,y) == 'D') {
            for(int i=0; i<animals.size(); i++) {
                if(animals.get(i).getPosX() == x && animals.get(i).getPosY() == y && animals.get(i).getCountProduct()>0) {
                    animals.get(i).getProduct();
                    inventory.add("GoatMilk");
                }
            }
         }
         // Sapi(C)
         else if(_c.getElement(x,y) == 'S') {
            for(int i=0; i<animals.size(); i++) {
                if(animals.get(i).getPosX() == x && animals.get(i).getPosY() == y && animals.get(i).getCountProduct()>0) {
                    animals.get(i).getProduct();
                    inventory.add("CowMilk");
                }
            }
         } else{
            System.out.println("Kosong");
         }   
    }

    /**
     * Inceract with facility.
     *
     * @param _f The Facility on Board.
     * @param side The side the player interact with.
     */
    public void interactFacility(Facility _f, int side) {
        if (isAdjacent(_f.getWellX(), _f.getWellY(), side)) {
            water = 10;
        } else if (isAdjacent(_f.getMixerX(), _f.getMixerY(), side)) {
            System.out.print("Bahan yang ingin dibuat : ");
            Scanner sc = new Scanner(System. in);
            String Bahan = sc. nextLine();
            if (Bahan.equals("Omelette")) {
                int Bahan1 = inventory.indexOf("ChickenEgg");
                if (Bahan1 == -1) {
                    System.out.println("Gaada ChickenEgg");
                }
                else {
                    int Bahan2 = inventory.indexOf("DuckEgg");
                    if (Bahan2 == -1) {
                        System.out.println("Gaada DuckEgg");
                    }
                    else {
                        Omelette a = new Omelette(1);
                        inventory.add("Omelette");
                        inventory.remove("ChickenEgg");
                        inventory.remove("DuckEgg");
                    }
                }
            }
            else if (Bahan.equals("Barbeque")) {
                int Bahan1 = inventory.indexOf("Pork");
                if (Bahan1 == -1) {
                    System.out.println("Gaada Pork");
                }
                else {
                    int Bahan2 = inventory.indexOf("LambMeat");
                    if (Bahan2 == -1) {
                        System.out.println("Gaada LambMeat");
                    }
                    else {
                        Barbeque a = new Barbeque(1);
                        inventory.add("Barbeque");
                        inventory.remove("Pork");
                        inventory.remove("LambMeat");
                    }
                }
            }
            else if (Bahan.equals("Sausage")) {
                int Bahan1 = inventory.indexOf("CowMeat");
                if (Bahan1 == -1) {
                    System.out.println("Gaada CowMeat");
                }
                else {
                    int Bahan2 = inventory.indexOf("GoatMeat");
                    if (Bahan2 == -1) {
                        System.out.println("Gaada GoatMeat");
                    }
                    else {
                        Sausage a = new Sausage(1);
                        inventory.add("Sausage");
                        inventory.remove("CowMeat");
                        inventory.remove("GoatMeat");
                    }
                }
            }
            else {
                System.out.println("Gaada SideProduct yang itu");
            }
        } else if (isAdjacent(_f.getTruckX(), _f.getTruckY(), side)) {
            if(inventory.isEmpty()) {
                System.out.println("Inventory Kosong");
            } else{
                String isi;
                while(!inventory.isEmpty()) {
                    isi = inventory.get(0);
                    inventory.remove(isi);
                    if ( isi.toString().equals("ChickenMeat")) {
                        money += 4;
                    } else if(isi.toString().equals("ChickenEgg")) {
                        money += 3;
                    } else if(isi.toString().equals("DuckMeat")) {
                        money += 5;
                    } else if(isi.toString().equals("DuckEgg")) {
                        money += 4;
                    } else if(isi.toString().equals("CowMeat")) {
                        money += 6;
                    } else if(isi.toString().equals("CowMilk")) {
                        money += 4;
                    } else if(isi.toString().equals("GoatMeat")) {
                        money += 5;
                    } else if(isi.toString().equals("GoatMilk")) {
                        money += 3;
                    } else if(isi.toString().equals("LambMeat")) {
                        money += 5;
                    } else if(isi.toString().equals("Omelette")) {
                        money += 3;
                    } else if(isi.toString().equals("Pork")) {
                        money += 5;
                    } else if(isi.toString().equals("Sausage")) {
                        money += 5;
                    } else if(isi.toString().equals("Barbeque")) {
                        money += 5;
                    }
                }
                System.out.println("Truck Succes");
            }
        }
    }

    /**
     * Kill animal.
     *
     * @param _c The Cell the Player refer to.
     * @param animals The list of animals on Board.
     */
    public void Kill(Cell _c, Vector<FarmAnimal> animals) {
            System.out.println("Select Side to Kill");
            System.out.println("1. Atas    2. Bawah");
            System.out.println("3. Kiri    4. Kanan");
            Scanner sc = new Scanner(System.in);
            int cc = sc.nextInt(); 
           switch(cc) {
               case 1 :
               {
                   cekHewan(posX-1, posY, _c);
    
                   for(int i=0; i<animals.size(); i++) {
                       if(animals.get(i).getPosX() == this.posX-1 && animals.get(i).getPosY() == this.posY) {
                           animals.get(i).setStatus(false);
                       }
                   }
                   break;
               }
               case 2 :
               {
                   cekHewan(posX+1, posY, _c);
    
                   for(int i=0; i<animals.size(); i++) {
                       if(animals.get(i).getPosX() == this.posX+1 && animals.get(i).getPosY() == this.posY) {
                           animals.get(i).setStatus(false);
                       }
                   }
                   break;
               }
               case 3 :
               {   
                   cekHewan(posX, posY-1, _c);
    
                   for(int i=0; i<animals.size(); i++) {
                       if(animals.get(i).getPosX() == this.posX-1 && animals.get(i).getPosY() == this.posY-1) {
                           animals.get(i).setStatus(false);
                       }
                   }
                   break;
               }
               case 4 :
               {
                   cekHewan(posX, posY+1, _c);
    
                   for(int i=0; i<animals.size(); i++) {
                       if(animals.get(i).getPosX() == this.posX && animals.get(i).getPosY() == this.posY+1) {
                           animals.get(i).setStatus(false);
                       }
                   }
                   break;
               }
               default :{
                System.out.println("Command Salah");
               }   
           }
            
       }

       /**
        * Check animal the player want to kill. If there is an animal,
        * the Player get the animal product. Otherwise, the player got nothing.
        *
        * @param x The X position of the cell Player interact with.
        * @param y The Y position of the cell Player interact with.
        * @param _c The cell the Player refer to.
        */
       public void cekHewan(int x, int y, Cell _c) {
           // ayam(A)
            if(_c.getElement(x,y) == 'A') {
                ChickenMeat a = new ChickenMeat();
                inventory.add("ChickenMeat");
                _c.setElement(x, y, ' ');
            }
            // bebek(B)
            else if(_c.getElement(x,y) == 'B') {
                DuckMeat b = new DuckMeat();
                inventory.add("DuckMeat");
                _c.setElement(x, y, ' ');
            } 
            // babi(b)
            else if(_c.getElement(x,y) == 'Z') {
                Pork c = new Pork();
                inventory.add("Pork");
                _c.setElement(x, y, ' ');
            } 
            // domba(D)
            else if(_c.getElement(x,y) == 'K') {
                LambMeat d = new LambMeat();
                inventory.add("LambMeat");
                _c.setElement(x, y, ' ');
            }
            // kambing(G)
            else if(_c.getElement(x,y) == 'D') {
                GoatMeat e = new GoatMeat();
                inventory.add("GoatMeat");
                _c.setElement(x, y, ' ');
            }
            // Sapi(C)
            else if(_c.getElement(x,y) == 'S') {
                CowMeat f = new CowMeat();
                inventory.add("CowMeat");
                _c.setElement(x, y, ' ');
            } else{
                System.out.println("Kosong");
            }   
            this.print();
       }

       /**
        * Grow grass on current position.
        *
        * @param _c The Cell the Player refer to.
        */
       public void grow(Cell _c) {
           if (water > 0) {
               // Mengisi petak cell dengan notasi rumput
               if(_c.getRumput(posX, posY) == ' ') {
                   _c.setRumput(posX, posY);
                   water-=1;
               }
               else {
                   // udah ada rumput, jadi do nothing/pesen error
                   System.out.println("Can't grow anymore!"); 
               }
           }
           else {
               // gaada air, sana pergi ke well
               System.out.println("insufficient water!");
           }
           this.print();
       }

}